### PR TITLE
[improvement](meta) Switch meta serialization to gson

### PR DESCRIPTION
**Pull request by iszhangpch**
## Proposed changes

Switch meta serialization to gson.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java [hunk 1]
```diff
@@ -1994,84 +2016,16 @@ private void setTableStateToNormal(Database db, boolean committed, boolean isRep
...
-        for (Map.Entry<String, String> entry : properties.entrySet()) {
-            Text.writeString(out, entry.getKey());
-            Text.writeString(out, entry.getValue());
+        if (Env.getCurrentEnvJournalVersion() < FeMetaVersion.VERSION_127) {
+            RestoreJob job = new RestoreJob();
+            job.readFields(in);
+            return job;
         }
+        String json = Text.readString(in);
+        return GsonUtils.GSON.fromJson(json, RestoreJob.class);
```
Comment (gavinchou) [inline fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java:2025]: Is `write()` missing?

Comment (iszhangpch) [reply fe/fe-core/src/main/java/org/apache/doris/backup/RestoreJob.java:2025]: No, I think it's not necessary to override write() of class AbstractJob.

**Comment by SWJTU-ZhangLei** [comment]
Can you add some ut for  Serialization and Deserialization, this is a demo in `TableAddOrDropColumnsInfoTest.java`

**Comment by iszhangpch** [comment]
> Can you add some ut for Serialization and Deserialization, this is a demo in `TableAddOrDropColumnsInfoTest.java`

**Comment by iszhangpch** [comment]
> Can you add some ut for Serialization and Deserialization, this is a demo in `TableAddOrDropColumnsInfoTest.java`

Yes

**Pull request closed without merging**